os/net/lwip/src/api/sockets.c

```diff
@@ -453,31 +468,36 @@ struct lwip_sock *trycheck_selwait_socket(int s)
  */
 int alloc_socket(struct netconn *newconn, int accepted)
 {
-	int i;
+	int idx;
+	struct socketlist *list;
+	struct lwip_sock *sock;
+
+	list = sched_getsockets();
+	if (!list) {
+		ndbg("could not get the socketlist\n");
+		return -1;
```

Comment:
maybe can use ERROR in place of -1

Reply:
or something similar

Reply:
when socket is not created then earlier they were returning -1 so i thought let's keep it same. seems no problem here.